erpnext/stock/doctype/purchase_receipt/purchase_receipt.py

```diff
@@ -180,6 +170,8 @@ def on_submit(self):
 
 		self.make_gl_entries()
 		self.update_package_tag_batch()
+		update_per_received_and_status_in_production_plan(self)
```

Comment:
@MyuddinKhatri same need to implement for cancel otherwise if you cancel purchase receipt it will not update the status of the production plan or it show you the material received.